### DISCUSSION TITLE
API reference with openapi schema

### DISCUSSION THREAD

**Discussion by baptisteArno** [shuding/nextra#1024] [Ideas]
Do you recommend anything that would allow me to create an API reference with nextra based on an openapi schema?

Otherwise, I might be interested in creating a plugin for that, what do you think?

**Comment by shuding** [comment]
Sounds like an interesting idea, can you write more about the plugin before start working on it? Like a Next.js plugin?

**Reply by baptisteArno** [reply]
Indeed, I'm not sure if it needs to have any relation with nextra?

The idea would be for the developer to just provide an `openapi.{yaml, json}` (and optionally other markdown pages) and it would generate a fully-fledge API doc with a playground. Much like: https://github.com/cloud-annotations/docusaurus-openapi.

Should it be something "native" to nextra or something else?

(Example of an openAPI schema: https://github.com/cloud-annotations/docusaurus-openapi/blob/main/demo/examples/petstore.yaml)

**Comment by naueramant** [comment]
This would be awesome to have 🙌

**Comment by sdairs** [comment]
@shuding https://vercel.com/docs/rest-api/endpoints <-- looks like Vercel is generating REST API docs by fetching the spec when you load the page and dynamically creating the pages. It would be pretty sweet if Nextra could do the same OOTB, perhaps there is internal work that could be re-used?

**Reply by sdairs** [reply]
Wrote more about it in this issue https://github.com/shuding/nextra/issues/1763
